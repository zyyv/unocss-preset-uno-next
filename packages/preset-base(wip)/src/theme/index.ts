/* @export-submodules */
export * from './colors'
export * from './default'
export * from './filters'
export * from './font'
export * from './misc'
export * from './screen'
export * from './types'
