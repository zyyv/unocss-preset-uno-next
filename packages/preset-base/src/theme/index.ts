/* @export-submodules */
export * from './colors'
export * from './default'
export * from './filters'
export * from './font'
export * from './misc'
export * from './size'
export * from './types'
