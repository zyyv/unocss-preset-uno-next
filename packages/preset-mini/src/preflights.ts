import type { Preflight } from '@unocss/core'
import type { PresetMiniOptions } from '.'
import type { Theme } from './theme'
import { entriesToCss, toArray } from '@unocss/core'

function basePreflight(options: PresetMiniOptions): Preflight<Theme> | undefined {
  return options.preflight
    ? {
        layer: 'preflights',
        getCSS({ theme, generator }) {
          if (theme.preflightBase) {
            let entries = Object.entries(theme.preflightBase)
            if (options.preflight === 'on-demand') {
              const keys = new Set(Array.from(generator.activatedRules).map(r => r[2]?.custom?.preflightKeys).filter(Boolean).flat())
              entries = entries.filter(([k]) => keys.has(k))
            }

            if (entries.length > 0) {
              let css = entriesToCss(entries)
              if (options.variablePrefix !== 'un-') {
                css = css.replace(/--un-/g, `--${options.variablePrefix}`)
              }
              const roots = toArray(theme.preflightRoot ?? ['*,::before,::after', '::backdrop'])
              return roots.map(root => `${root}{${css}}`).join('')
            }
          }
        },
      }
    : undefined
}

export function preflights(options: PresetMiniOptions): Preflight<Theme>[] | undefined {
  return ([
    basePreflight(options),
  ] as (Preflight<Theme> | undefined)[]).filter(Boolean) as Preflight<Theme>[]
}
