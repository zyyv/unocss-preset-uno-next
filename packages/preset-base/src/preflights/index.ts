import type { Preflight } from '@unocss/core'
import type { Theme } from '../theme/types'
import { animate } from './animate'
import { theme } from './theme'

export const preflights: Preflight<Theme>[] = [
  theme(),
  animate(),
]
