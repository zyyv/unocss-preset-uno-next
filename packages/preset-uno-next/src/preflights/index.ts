import type { Preflight } from '@unocss/core'
import type { PresetUnoNextOptions } from '..'
import type { Theme } from '../theme/types'
import { theme } from './theme'

export const preflights: (options: PresetUnoNextOptions) => Preflight<Theme>[] = (options) => {
  return [
    theme(options),
  ]
}
