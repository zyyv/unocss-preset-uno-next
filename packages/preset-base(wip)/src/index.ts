import { preflights } from './preflights'
import { rules } from './rules/default'
import { theme } from './theme'

export function presetBase() {
  return {
    name: 'preset-base',
    rules,
    theme,
    preflights,
  }
}
