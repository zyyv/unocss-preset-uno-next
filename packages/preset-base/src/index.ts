import { preflights } from './preflights'
import { theme } from './theme'

export function presetBase() {
  return {
    name: 'preset-base',
    theme,
    preflights,
  }
}
