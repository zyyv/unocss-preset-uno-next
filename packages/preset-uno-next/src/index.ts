import type { PresetOptions } from '@unocss/core'
import { preflights } from './preflights'
import { rules } from './rules/default'
import { theme } from './theme'

export interface DarkModeSelectors {
  /**
   * Selector for light variant.
   *
   * @default '.light'
   */
  light?: string

  /**
   * Selector for dark variant.
   *
   * @default '.dark'
   */
  dark?: string
}

export interface PresetMiniOptions extends PresetOptions {
  /**
   * Dark mode options
   *
   * @default 'class'
   */
  dark?: 'class' | 'media' | DarkModeSelectors
  /**
   * Generate tagged pseudo selector as `[group=""]` instead of `.group`
   *
   * @default false
   */
  attributifyPseudo?: boolean
  /**
   * Prefix for CSS variables.
   *
   * @default 'un-'
   */
  variablePrefix?: string
  /**
   * Utils prefix. When using tagged pseudo selector, only the first truthy prefix will be used.
   *
   * @default undefined
   */
  prefix?: string | string[]
  /**
   * Generate preflight
   *
   * @default true
   */
  preflight?: boolean | 'on-demand'

  /**
   * Enable arbitrary variants, for example `<div class="[&>*]:m-1 [&[open]]:p-2"></div>`.
   *
   * Disable this might slightly improve the performance.
   *
   * @default true
   */
  arbitraryVariants?: boolean
}

export function presetUnoNext() {
  return {
    name: 'preset-base',
    rules,
    theme,
    preflights,
  }
}
