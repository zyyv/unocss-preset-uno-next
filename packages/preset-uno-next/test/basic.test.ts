import { createGenerator, presetUno } from 'unocss'
import { expect, it } from 'vitest'
import { presetUnoNext } from '../src/index'
import { presetMiniTargets } from './assets/preset-mini-targets'

it('presetStarter', async () => {
  const unoNext = await createGenerator({
    presets: [presetUnoNext()],
  })

  const uno = await createGenerator({
    presets: [presetUno()],
  })

  const needFixed = []

  let sameString = `## Same Tokens

| Token | Same | UnoNext | Uno |
| --- | --- | --- | --- |
  `

  let differentString = `
## Different Tokens

| Token | Same | UnoNext | Uno |
| --- | --- | --- | --- |
`

  let result = `# Overview

`

  for (const target of presetMiniTargets) {
    const [cssnext, css] = await Promise.all([
      unoNext.generate(target, { preflights: false }).then(r => r.css.replace('/* layer: default */', '').trim()),
      uno.generate(target, { preflights: false }).then(r => r.css.replace('/* layer: default */', '').trim()),
    ])

    if (css && !cssnext) {
      needFixed.push(target)
    }

    const escape = (code: string) => `<code>${code.replace(/\|/g, '\\|').replace(/\n/g, '<br>')}</code>`

    const same = cssnext === css

    if (same) {
      sameString += `| \`${target}\` | ✅ | ${cssnext ? `${escape(cssnext)}` : '❓'} | ${css ? `${escape(css)}` : '❓'} |\n`
    }
    else {
      differentString += `| \`${target}\` | ❌ | ${cssnext ? `${escape(cssnext)}` : '❓'} | ${css ? `${escape(css)}` : '❓'} |\n`
    }
  }

  result += sameString + differentString

  expect(result).toMatchFileSnapshot('./fixtures/token-different.test.md')
  expect(needFixed).toMatchSnapshot()
})

it('test case', async () => {
  const unoNext = await createGenerator({
    presets: [presetUnoNext()],
  })

  const { css } = await unoNext.generate('outline-color-red-100', { preflights: false })

  expect(css).toMatchInlineSnapshot(`
    "/* layer: default */
    .outline-color-red-100{--un-outline-color-opacity:100%;outline-color:color-mix(in oklch, var(--color-red-100) var(--un-outline-color-opacity), transparent);}"
  `)
})
