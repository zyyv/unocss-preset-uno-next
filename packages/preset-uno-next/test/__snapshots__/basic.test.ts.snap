// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`presetStarter 1`] = `
[
  "outline-size-none",
  "outline-color-red-1",
  "outline-blue-2",
  "border-size-none",
  "border-red2",
  "color-blue-gray",
  "color-blue-gray-400",
  "color-bluegray",
  "color-bluegray-400",
  "color-blue-gray-400/10",
  "color-blue-gray/10",
  "color-bluegray-400/10",
  "color-bluegray/10",
  "text-red2",
  "ring-red2",
  "ring-red2/5",
  "ring-width-px",
  "ring-size-px",
  "ring-offset-red2",
  "ring-offset-red2/5",
  "decoration-size-none",
  "decoration-offset-none",
  "basis-none",
  "gap-none",
  "inset-none",
  "ring-none",
  "ring-offset-none",
  "ring-offset-green5",
  "p-none",
  "m-none",
  "p-ie-none",
  "m-inline-none",
  "pb",
  "px",
  "p-x",
  "pxy",
  "p-xy",
  "pis",
  "p-is",
  "m-block",
  "mbs",
  "mxy",
  "m-xy",
  "stroke-size-none",
  "stroke-offset-none",
  "preserve-3d",
  "preserve-flat",
  "indent",
  "indent-lg",
  "media-mouse:block",
  "group-hover:media-mouse:bg-red",
  "placeholder-color-red-1",
  "hover:not-first:checked:bg-true-gray/10",
  "hover:is-first:checked:bg-true-gray/10",
]
`;
